docs/Developers/Git_Basics.rst

```diff
@@ -178,6 +226,17 @@ and your commit will be pushed online.
 Creating a Pull Request
 -----------------------
 
+If you've made it this far, and you're still using the 'master' branch, then
+we're definitely going to be able to tell you have not been reading this
+documentation. Naughty, naughty, but there is still a way to fix your changes
+if you have already commited. You can run the following command, which will
+take you to a new branch containing all of your commited changes (Note: Some
+sample naming conventions can be found under the "Creating a branch" section).
+
+::
+
+    $ git checkout -b <branchname>
```

Comment:
then perhaps link to the common git issues section for info on how to reset master to pristine state